File: spring-server/app/src/main/java/tournament/model/Song.java
package tournament.model;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

@Entity
@Getter @Setter @NoArgsConstructor @ToString @EqualsAndHashCode
public class Song {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;
    private String title;
    private String artist;
    private String spotify;
    private String youtube;
}

File: react-server/src/model/Song.ts
export default interface Song {
    id: number;
    title: string;
    artist: string;
    spotify?: string;
    youtube?: string;
    activeRound?: boolean;
}


export interface BracketSong extends Song {
    parent1VoteCount?: number;
    parent2VoteCount?: number;
    receivedVoteCount?: number;
    totalVoteCount?: number;
}